multiplayer/src/services/gameClient.ts

```diff
@@ -231,14 +232,15 @@ class GameClient {
         console.log(
             `Server said we're joined as "${msg.role}" in slot "${msg.slot}"`
         );
-        const gameMode = msg.gameMode;
+        const { gameMode, shareCode } = msg;
 
-        setGameModeAsync(gameMode, msg.slot);
+        // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+        (await setGameMetadataAsync(shareCode)) && (await setGameModeAsync(gameMode, msg.slot));
```

Comment:
I guess this eslint rule is trying to push you towards doing a conditional here

```suggestion
        if (await setGameMetadataAsync(shareCode)) 
            await setGameModeAsync(gameMode, msg.slot);
```

which is maybe a little easier to parse? doesn't really matter at all

fwiw it has helped me remember to delete accidental one liners where I was exploring api with intellisense a few times (e.g. https://github.com/microsoft/pxt/pull/9132/commits/2f4f33768c9f568c8446f60cacccb57ad3260595)